### PR TITLE
Invoker refactor 2

### PR DESCRIPTION
**Pull request by janezpodhostnik**
## Goal of this PR

Instead of reusing the flow-go code to setup the FVM I opted for copying it over, since reuse is not that simple currently. 

The next step would be to change flow-go to allow for the reuse of the logic that sets up QueryExecutor.

I left some todos in there for logging and reporting, I'll address that next.

### Misc

- [ ] PR title will be clear as part of the changelog
- [ ] PR is against the correct branch
- [ ] PR is labelled appropriately
- [ ] PR is linked to an issue

### CODE REVIEW

File: service/invoker/blocks.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package invoker
+
```
Comment (koko1123) [inline service/invoker/blocks.go:2]: thanks for this! any other `/env` structures that need to be here?

File: service/invoker/invoker.go [hunk 2]
```diff
@@ -55,17 +47,57 @@ func New(index archive.Reader, options ...func(*Config)) (*Invoker, error) {
...
+		fvm.WithReusableCadenceRuntimePool(
+			reusableRuntime.NewReusableCadenceRuntimePool(
+				computation.ReusableCadenceRuntimePoolSize,
+				runtime.Config{
+					TracingEnabled:        cfg.CadenceTracing,
+					AccountLinkingEnabled: true,
+					// Attachments are enabled everywhere except for Mainnet
+					AttachmentsEnabled: chainID != flow.Mainnet,
+					// Capability Controllers are enabled everywhere except for Mainnet
+					CapabilityControllersEnabled: chainID != flow.Mainnet,
```
Comment (koko1123) [inline service/invoker/invoker.go:73]: should we leave a `todo` here for when mainnet has these? I'm assuming they'll appear after the HCU?

**Review by koko1123 — COMMENTED** [review]
couple of questions but looks good to me otherwise!

File: service/invoker/invoker.go (hunk 2 above)
Comment (janezpodhostnik) [reply service/invoker/invoker.go:73]: I added a comment above, saying where this is copied from. When that changes, this should change too. Ideally we would use the same code for both, and that will be the next step.

File: service/invoker/blocks.go (hunk 1 above)
Comment (janezpodhostnik) [reply service/invoker/blocks.go:2]: not sure, I think this is all for now.

**Review by koko1123 — APPROVED** [review]

**Pull request merged**